minio/credentials/providers.py

```diff
@@ -504,22 +504,18 @@ def retrieve(self) -> Credentials:
             token = res.data.decode("utf-8")
             headers = {"X-aws-ec2-metadata-token": token} if token else None
 
+            url += "/latest/meta-data/iam/security-credentials/"
             # Get role name
             res = _urlopen(
                 self._http_client,
                 "GET",
-                urlunsplit(
-                    url_replace(
-                        urlsplit(url),
-                        path="/latest/meta-data/iam/security-credentials/",
-                    ),
-                ),
-                headers=headers,
+                url,
+                headers=headers
             )
             role_names = res.data.decode("utf-8").split("\n")
             if not role_names:
                 raise ValueError(f"no IAM roles attached to EC2 service {url}")
-            url += "/" + role_names[0].strip("\r")
+            url += role_names[0].strip("\r")
```

Comment:
The role gets added at the end of the URL that was updated on L507.

Reply:
This is also in the Go implementation here: https://github.com/minio/minio-go/blob/e337e77c06bbf8a65247fb018e59c1263e4d86e1/pkg/credentials/iam_aws.go#L400-L402

We just missed it in the revisions in https://github.com/minio/minio-py/pull/1437/.